pkg/cleaner/dnats.go

```diff
@@ -0,0 +1,78 @@
+package cleaner
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/go-logr/logr"
+	"github.com/vmware/cloud-provider-for-cloud-director/pkg/vcdsdk"
+	capvcd "github.com/vmware/cluster-api-provider-cloud-director/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DNATCleaner struct {
+	cli client.Client
+}
+
+func NewDNATCleaner(cli client.Client) *DNATCleaner {
+	return &DNATCleaner{cli: cli}
+}
+
+// force implementing Cleaner interface
+var _ Cleaner = &DNATCleaner{}
+
+func (lbc *DNATCleaner) Clean(ctx context.Context, log logr.Logger, vcdClient *vcdsdk.Client, c *capvcd.VCDCluster) (bool, error) {
+	log = log.WithName("DNATCleaner")
+	gateway, err := vcdsdk.NewGatewayManager(ctx, vcdClient, c.Spec.OvdcNetwork, c.Spec.LoadBalancerConfigSpec.VipSubnet)
```

Comment:
Can we move this logic to a common place? We use it for all cleaners.

Reply:
done in https://github.com/giantswarm/cluster-api-cleaner-cloud-director/pull/9/commits/de3a0e67d9d6cf4da4b76bfa0a4afea2920d4a1c

Reply:
Thanks